src/olympia/addons/serializers.py

```diff
@@ -513,24 +514,41 @@ def _check_for_existing_versions(self, version_string):
                 {'version': msg.format(version_string=version_string)}
             )
 
+    def _check_is_greater_version_number(self, version_string):
+        if (
+            previous_version := self.addon.current_version
+        ) and previous_version.version >= version_string:
+            msg = gettext(
+                'Version {version_string} must be greater than the previous approved '
+                'version {previous_version_string}.'
+            )
+            raise exceptions.ValidationError(
+                {
+                    'version': msg.format(
+                        version_string=version_string,
+                        previous_version_string=previous_version.version,
+                    )
+                }
+            )
+
     def validate(self, data):
         if not self.instance:
             guid = self.addon.guid if self.addon else self.parsed_data.get('guid')
-            self._check_blocklist(guid, self.parsed_data.get('version'))
+            version_string = VersionString(self.parsed_data.get('version'))
+            self._check_blocklist(guid, version_string)
             if self.addon:
-                self._check_for_existing_versions(self.parsed_data.get('version'))
-
-                # Also check for submitting listed versions when disabled.
-                if (
-                    data['upload'].channel == amo.CHANNEL_LISTED
-                    and self.addon.disabled_by_user
-                ):
-                    raise exceptions.ValidationError(
-                        gettext(
-                            'Listed versions cannot be submitted while add-on is '
-                            'disabled.'
+                self._check_for_existing_versions(version_string)
+
+                if data['upload'].channel == amo.CHANNEL_LISTED:
+                    self._check_is_greater_version_number(version_string)
```

Comment:
Why are we limiting it to listed versions? Wouldn't this be a valuable check for unlisted versions too? Or do we just not care about those?

Reply:
the issue requested just listed versions have the restriction applied.